packages/sdk/src/swap/sdk.ts

```diff
@@ -128,9 +137,7 @@ export class SwapSDK {
   }
 
   private async getStateChainEnvironment(): Promise<Environment> {
-    this.stateChainEnvironment ??= await getEnvironment(this.rpcConfig);
-
-    return this.stateChainEnvironment;
+    return this.stateChainEnvironmentCache.get('cf_environment');
```

Comment:
can we have a test for this?

Reply:
i wrote in the description. i tried for too long and failed. Something about mocks not resetting